octopoes/connectors/services/xtdb.py

```diff
@@ -0,0 +1,171 @@
+"""XTDB HTTP client."""
+
+from __future__ import annotations
+
+import logging
+from datetime import timezone, datetime
+from enum import Enum
+from http import HTTPStatus
+from typing import Union, Optional, Tuple, List, Dict, Any, Type, Callable, cast
+
+import requests
+from pydantic import BaseModel, Field
+from requests import Response, HTTPError
+
+logger = logging.getLogger(__name__)
+
+
+class OperationType(Enum):
+    """Operation types for XTDB transactions."""
+
+    PUT = "put"
+    DELETE = "delete"
+    MATCH = "match"
+    EVICT = "evict"
+    FN = "fn"
+
+
+Operation = Tuple[OperationType, Union[str, Dict[str, Any]], Optional[datetime]]
+
+
+class Transaction(BaseModel):
+    """A XTDB transaction, consisting of a list of operations."""
+
+    operations: List[Operation] = Field(alias="tx-ops")
+
+    class Config:
+        """Serialize operations as 'tx-ops' in JSON."""
+
+        allow_population_by_field_name = True
+
+
+class XTDBHTTPSession(requests.Session):
+    """A requests session that adds the base URL to all requests."""
+
+    def __init__(self, base_url: str):
+        """Initialize instance."""
+        super().__init__()
+
+        self._base_url = base_url
+        self.headers["Accept"] = "application/json"
+
+    def request(self, method: str, url: Union[str, bytes], **kwargs) -> requests.Response:  # type: ignore
+        """Execute request with prepended base URL."""
+        return super().request(method, self._base_url + str(url), **kwargs)
+
+
+class XTDBStatus(BaseModel):
```

Comment:
Prefer to use python names and use field aliases

Reply:
Fixed